packages/core-transaction-pool/lib/guard.js

```diff
@@ -187,8 +187,9 @@ module.exports = class TransactionGuard {
   async __determineExcessTransactions () {
     for (let transaction of this.broadcast) {
       const hasExceeded = await this.pool.hasExceededMaxTransactions(transaction)
+      const hasVote = this.pool.checkIfSenderHasVoteTransactions(transaction.senderPublicKey)
 
-      if (hasExceeded) {
+      if (hasExceeded || hasVote) {
```

Comment:
If a vote is already present it should be rejected as invalid, not excess.

Reply:
Excess in this context means that the limit per request was exceeded, meaning throttling kicked in.

Exceeding any other arbitrary limits that are exceeded should be handled as invalid and return an error together with their ID so the user can be informed about why the transaction was rejected.

In your case you would mark the vote transaction as invalid and return an error like `[address/publicKey] already has a pending vote`.